src/main/kotlin/io/emeraldpay/dshackle/upstream/calls/EthereumCallSelector.kt

```diff
@@ -68,9 +68,9 @@ class EthereumCallSelector(
             return blockTagSelector(params, 2, null, head)
         } else if (method in DefaultEthereumMethods.withFilterIdMethods) {
             return sameUpstreamMatcher(params)
-        } else if (method in GET_BY_HASH_OR_NUMBER_METHODS) {
+        } else if (!passthrough && method in GET_BY_HASH_OR_NUMBER_METHODS) {
             return blockMethodSelector(method, params, head)
-        } else if (method == "eth_getLogs") {
+        } else if (!passthrough && method == "eth_getLogs") {
```

Comment:
maybe create one big block with passthrough check and write a comment about purpose of this check?

Reply:
I fixed it, but don't understand if the order of the conditions is important.